test/dom/dom.ts

```diff
@@ -0,0 +1,493 @@
+   it('load() Clob as any', () => {
+      let c: any = ion.load('month::{{"February"}}')!;
+
+      assert.equal(IonTypes.CLOB, c.getType());
+      assert.deepEqual(['month'], c.getAnnotations());
+
+      assert.deepEqual(encodeUtf8("February"), c);
+   });
+
+   it('load() Blob as Value', () => {
+      let b: Value = ion.load('quote::{{VG8gaW5maW5pdHkuLi4gYW5kIGJleW9uZCE=}}')!;
+
+      assert.equal(IonTypes.BLOB, b.getType());
+      assert.deepEqual(['quote'], b.getAnnotations());
+
+      assert.deepEqual(encodeUtf8("To infinity... and beyond!"), b.uInt8ArrayValue());
+   });
+
+   it('load() Blob as any', () => {
+      let b: any = ion.load('quote::{{VG8gaW5maW5pdHkuLi4gYW5kIGJleW9uZCE=}}')!;
+
+      assert.equal(IonTypes.BLOB, b.getType());
+      assert.deepEqual(['quote'], b.getAnnotations());
+
+      assert.deepEqual(encodeUtf8("To infinity... and beyond!"), b);
+   });
+
+   it('load() List as Value', () => {
+      let l: Value = ion.load('planets::["Mercury", "Venus", "Earth", "Mars"]')!;
+
+      assert.equal(IonTypes.LIST, l.getType());
+      assert.deepEqual(['planets'], l.getAnnotations());
+
+      // Iteration
+      for (let planet of l.values()!) {
+         assert.isNotNull(planet.stringValue());
+      }
+
+      // Indexing
+      assert.equal("Mercury", l.get(0)!.stringValue());
+      assert.equal("Venus", l.get(1)!.stringValue());
+      assert.equal("Earth",l.get(2)!.stringValue());
+      assert.equal("Mars", l.get(3)!.stringValue());
+
+      let planets: string[] = [...l.values()!].map(s => s.stringValue()!);
+      assert.equal(4, planets.length);
+   });
+
+   it('load() List as any', () => {
+      let l: any = ion.load('planets::["Mercury", "Venus", "Earth", "Mars"]')!;
+
+      assert.equal(IonTypes.LIST, l.getType());
+      assert.deepEqual(['planets'], l.getAnnotations());
+
+      // Iteration
+      for (let planet of l) {
+         assert.isTrue(planet instanceof String);
+      }
+
+      // Indexing
+      assert.equal("Mercury", l[0]);
+      assert.equal("Venus", l[1]);
+      assert.equal("Earth", l[2]);
+      assert.equal("Mars", l[3]);
+
+      assert.equal(4, l.length);
+
+      // Directly use Array methods
+      assert.isTrue(l.findIndex(s => s == "Earth") > 0);
+      assert.isNotNull(l.find(s => s == "Earth"));
+   });
+
+   it('load() SExpression as Value', () => {
+      let s: Value = ion.load('planets::("Mercury" "Venus" "Earth" "Mars")')!;
+
+      assert.equal(IonTypes.SEXP, s.getType());
+      assert.deepEqual(['planets'], s.getAnnotations());
+
+      // Iteration
+      for (let planet of s.values()!) {
+         assert.isNotNull(planet.stringValue());
+      }
+
+      // Indexing
+      assert.equal("Mercury", s.get(0)!.stringValue());
+      assert.equal("Venus", s.get(1)!.stringValue());
+      assert.equal("Earth",s.get(2)!.stringValue());
+      assert.equal("Mars", s.get(3)!.stringValue());
+
+      let planets: string[] = [...s.values()!].map(s => s.stringValue()!);
```

Comment:
simplify to: `[...s.values()!] --> s.values()!`  ?

Reply:
`s.values()` returns an iterator and doesn't have a `map` method.